lib/mllp/sender.ex

```diff
@@ -391,6 +404,9 @@ defmodule MLLP.Sender do
     %State{state | pending_reconnect: ref}
   end
 
+  defp maybe_close_socket(%State{socket: nil}), do: :ok
+  defp maybe_close_socket(state), do: state.tcp.close(state.socket)
```

Comment:
This function is similar to `stop_connection`. I also noticed, `stop_connection` is being called in reconnect function but we never reconnect after that. May be a separate bug